python/cudf/cudf/core/column/categorical.py

```diff
@@ -104,7 +104,7 @@ def __init__(self, parent: SeriesOrSingleColumnIndex):
         super().__init__(parent=parent)
 
     @property
-    def categories(self) -> "cudf.core.index.BaseIndex":
+    def categories(self) -> "cudf.core.index.GenericIndex":
```

Comment:
Is this always `CategoricalIndex`?

Reply:
The `.categories` method accepts  `CategoricalAccesor` or `SingleColumnCategoricalIndex` and returns either of the following:
- `Index`   (inherits from `BaseIndex`) for categories with  object dtypes 
- `NumericIndex`  - `Int64Index`, `Float64Index ` (inherits from `GenericIndex`) for categories with numeric dtypes

Reply:
Accepts `CategoricalAccesor` or `SingleColumnCategoricalIndex` and returns the following: 

- `Index` (pandas) or  `StringIndex` (cudf)  >> object dtypes

- `Int64Index` or `Float64Index`  >> numeric dtypes

- `FloatIndex` (pandas) and `StringIndex` (cudf) >> None dtypes

- `Index` (pandas) and `GenericIndex` (cudf)  >> boolean  dtypes

Reply:
I'm confused. This is a property of `CategoricalAccessor`, right? What do you mean by "accepts"? Isn't the accessor always created via `Series.cat`?

Reply:
Apologies for the confusion, I didn't realize that` CategoricalIndex `has its own `.categories `method (separate from the `CategoricalAccessor`  class)